airflow/configuration.py

```diff
@@ -580,10 +589,24 @@ def getint(self, section, key):
     def getfloat(self, section, key):
         return float(self.get(section, key))
 
+    def remove_option(self, section, option):
```

Comment:
I don't think that we want to override `remove_option`. Allowing `remove_option` to cascade to `self.defaults` muddies the difference between set options and default options.